include/stdio.h

```diff
@@ -164,7 +164,7 @@ FAR char *gets(FAR char *s);
 FAR char *gets_s(FAR char *s, rsize_t n);
 void   rewind(FAR FILE *stream);
 
-#ifndef CONFIG_STDIO_DISABLE_BUFFERING
+#if !defined(CONFIG_STDIO_DISABLE_BUFFERING) || defined(CONFIG_LIBCXX)
```

Comment:
should we always set CONFIG_STDIO_DISABLE_BUFFERING to n when CONFIG_LIBCXX equal y? since libcxx call setbuf in iostream class.

Reply:
while i haven't investigated those usage deeply, i guess they are not "mandatory" parts of the library as helloxx/cxxtest are working for me.

Reply:
Since libcxx will increase at least 20KB code, I don't think that it's benefit to disable CONFIG_STDIO_DISABLE_BUFFERING

Reply:
while i don't know how much CONFIG_STDIO_DISABLE_BUFFERING can save, i guess every bytes can accumulate.

Reply:
Why don't let the linker remove the unused code for us? and it's also strange to couple setbuf/setvbuf with CONFIG_LIBCXX.

Reply:
btw, is there a reasonable way to make "always set CONFIG_STDIO_DISABLE_BUFFERING to n when CONFIG_LIBCXX equal y" in Kconfig?
"select" looks similar but it doesn't seem to allow "select !STDIO_DISABLE_BUFFERING".
"depends on !STDIO_DISABLE_BUFFERING" sounds too strong for this in my taste.

Reply:
How about this change:
https://github.com/apache/incubator-nuttx/pull/2119
since libc always provide a dummy implementaion:
https://github.com/apache/incubator-nuttx/blob/master/libs/libc/stdio/lib_setbuf.c#L85
https://github.com/apache/incubator-nuttx/blob/master/libs/libc/stdio/lib_setvbuf.c#L275
the correct fix is always define the prototype and include ib_setbuf.c/ib_setvbuf.c into build.

Reply:
@xiaoxiang781216 that PR is now merged.